group-coordinator/src/main/java/org/apache/kafka/coordinator/group/common/TopicIdToPartition.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.common;
+
+import org.apache.kafka.common.Uuid;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class TopicIdToPartition {
+    private final Uuid topicId;
+    private final Integer partition;
+    private final Optional<List<String>> rackIds;
+
+    public TopicIdToPartition(Uuid topicId, Integer topicPartition, Optional<List<String>> rackIds) {
+        this.topicId = Objects.requireNonNull(topicId, "topicId can not be null");
+        this.partition = Objects.requireNonNull(topicPartition, "topicPartition can not be null");
+        this.rackIds = rackIds;
```

Comment:
this should also be not null. If rackIds is empty, we should pass in Optional.empty()

Reply:
ohh okay got it! thanks!